src/models/memory-behavior.js

```diff
@@ -0,0 +1,20 @@
+const Behavior = require('./behavior');
+const utils = require('../utils');
+
+// TODO(@ryzokuken): Switch to a more persistent memory model later
+// (Hint: Databases)
+let memory = [];
+
+class MemoryBehavior extends Behavior {
+  constructor(trigger, action, keyword) {
+    const newAction = (botNick, username, ...args) => {
+      if (!utils.contains(memory, username)) {
+        memory.push(username);
```

Comment:
Could we use a key value store here like `memory.get('key')` and `memory.set('key', 'value')`? That way we could also pass in an option for a function of what to remember - we could remember for example what time they were first remembered, or other things we haven't thought of yet. It seems more scalable to problems we want to solve soon like "responding only i to newcomers" and is closer to a key/value database model anyways. What do you think?

Reply:
Later implementing db w something like this:

https://preview.npmjs.com/package/key-value-store

Reply:
Ok or maybe https://www.npmjs.com/package/interface-datastore

Reply:
@jywarren you up for a really quick chat?